### PR TITLE
Prevent unnecessary retries in the LRP instances client

### PR DESCRIPTION
**Pull request by cwlbraa**
Prior to this change, the LRP instances client would retry repeatedly if you asked for instances of STOPPED processes via the various `/stats` endpoints. Instead, this change confirms that Eirini is not running stopped apps. In addition, this change will retry any non-404 error. We think this might help with reliability in large clusters.

tracker story: https://www.pivotaltracker.com/story/show/175381591

vmware-internal report of the bug: https://vmware.slack.com/archives/C01BQNTR9RT/p1603226144071200

### CODE REVIEW

File: spec/unit/lib/cloud_controller/opi/instances_client_spec.rb [hunk 1]
```diff
@@ -12,7 +12,12 @@
     )
   end
 
-  let(:process) { double(guid: 'my-process-guid', version: 'my-version-guid') }
+  let(:process) { VCAP::CloudController::ProcessModel.make(
```
Comment (tcdowney) [inline spec/unit/lib/cloud_controller/opi/instances_client_spec.rb:15]: nit: this indentation is a little strange

File: lib/cloud_controller/opi/instances_client.rb [hunk 2]
```diff
@@ -34,20 +38,11 @@ def ==(other)
...
-        handle_error(resp_json)
-      rescue CloudController::Errors::NoRunningInstances => e
-        sleep(1)
-        retry if (retries += 1) < 5
-        raise e
-      end
-      process_guid = resp_json['process_guid']
-      resp_json['instances'].map do |instance|
-        ActualLRP.new(instance, process_guid)
+      return confirm_stopped(process) if process.stopped?
```
Comment (tcdowney) [inline lib/cloud_controller/opi/instances_client.rb:41]: What calls this method? I'm wondering why it's trying to even get instances for a stopped process?

File: spec/unit/lib/cloud_controller/opi/instances_client_spec.rb (hunk 1 above)
Comment (cwlbraa) [reply spec/unit/lib/cloud_controller/opi/instances_client_spec.rb:15]: i blame rubocop

**Comment by sweinstein22** [comment]
Hi @cwlbraa , @JenGoldstrich and I are working on moving our default branch from master to main, so we are going to close this issue out. If you could please reopen it against the new main branch we'd appreciate it.

Thanks,
Sarah and Jenna

File: lib/cloud_controller/opi/instances_client.rb (hunk 2 above)
Comment (tcdowney) [reply lib/cloud_controller/opi/instances_client.rb:41]: cool sounds good 👍

**Pull request merged**